optimade/validator/validator.py

```diff
@@ -798,7 +802,8 @@ def _test_info_or_links_endpoint(self, request_str: str) -> Union[bool, Any]:
                 request=request_str,
             )
             if not deserialized:
-                return response
+                return response.json()
```

Comment:
This might fail, I guess, if the response is not a JSON. This could be the case for `/versions` or if the response is `404 Not Found`.
You could perhaps wrap it in a try/except or something?
Or perhaps I'm missing something that this is essentially already checked. Perhaps in `_get_endpoint()`?

Reply:
The control flow is a bit messy here, but `_deserialize_response` catches the JSON errors and raises the validation warning in this case. The versions endpoint, as the one non-JSON endpoint, is checked explicitly with its own separate function so don't think there's an issue here.

Reply:
Okay. Well just for a tiny bit of clarity, could you add a comment here then that JSON-serialization checks have been performed in `_deserialize_response()`?

Reply:
I think more comments in general will help us to see where might optimize and streamline this code further in the future?
Also, it should hopefully make it easier to more quickly debug, although too many comments might be a bit too overwhelming. There's a golden ratio there somewhere 😅

Reply:
See comment above, think I've done it in a cleaner way now so hopefully less confusion. Added a few comments to the top level flow too